### ISSUE TITLE
Error Message: not a go package or not in GOPATH

### ISSUE THREAD

**Issue by RenEvo** [kardianos/govendor#169]
This is plaguing me pretty hardcore at the moment, as it is happening on both windows and mac for me, quite possibly I am just being stupidly noob here, but...

```
export GOPATH=`pwd`
go get github.com/kardianos/govendor
./bin/govendor init
```

I get the error above. When validating with go env, the working directory IS the GOPATH.

```
go env
GOARCH="amd64"
GOBIN=""
GOEXE=""
GOHOSTARCH="amd64"
GOHOSTOS="darwin"
GOOS="darwin"
GOPATH="/Users/projects/go-sandbox"
GORACE=""
GOROOT="/usr/local/go"
GOTOOLDIR="/usr/local/go/pkg/tool/darwin_amd64"
GO15VENDOREXPERIMENT="1"
CC="clang"
GOGCCFLAGS="-fPIC -m64 -pthread -fno-caret-diagnostics -Qunused-arguments -fmessage-length=0 -fno-common"
CXX="clang++"
CGO_ENABLED="1"
```

```
../bin/govendor init
Error: Package "/Users/projects/go-sandbox" not a go package or not in GOPATH.
```

Pulling my hair out on this, any insight?

**Comment by RenEvo** [comment]
Tracking it down to possibly here:
https://github.com/kardianos/govendor/blob/master/internal/pathos/path.go#L25

**Comment by RenEvo** [comment]
As I have seen a lot of others do, I added a bunch of printf lines everywhere in the code to try and track this down.

What I found is that internally the [gopaths are appended with /src/](https://github.com/kardianos/govendor/blob/e1feb5e23119ff2ee2d0e7d8d3dce8cd308851fe/context/context.go#L206), I am sure this is for a good reason, however, when comparing, it compares against the [working directory](https://github.com/kardianos/govendor/blob/e1feb5e23119ff2ee2d0e7d8d3dce8cd308851fe/context/context.go#L129). This causes issues, as seen here when doing an output on the ./src/ directory inside my GOPATH.

```
input : /users/projects/go-sandbox/src
prefix: /users/projects/go-sandbox/src/
```

Input is my working directory, prefix is the generated based on the GOPATH + /src/.

I really think I must be doing something wrong here, or this is just a weird edge case bug I am hitting?

Also, if it is expected that we run govendor from the ./src/ folder, the documentation should be updated to reflect this, caused TONS of confusion for me.

**Comment by kaey** [comment]
Go's standard directory layout looks like:
```
$GOPATH/src/project1
$GOPATH/src/project2
$GOPATH/src/github.com/me/myproject
...
etc
```

Govendor follows this structure. There aren't any reasons to run govendor in src directory (at least I can't think of any)

**Comment by kardianos** [comment]
@RenEvo What is the error message that would make more sense to you in this case?

**Comment by sean-** [comment]
I'm running into the same problem intermittently as well and it's driving me crazy.  I seem to run into this when running `govendor vet +local $(VETARGS)` and it's non-deterministic.  I'll be working and suddenly I'll see:

```
11:01 seanc@hormesis % make vet
--> Running govendor vet -asmdecl -atomic -bool -buildtags -copylocks -methods  -nilfunc -printf -rangeloops -shift -structtags -unsafeptr +local
Error: Package "/usr/home/seanc/go/src/github.com/hashicorp/nomad" not a go package or not in GOPATH.
*** Error code 2

Stop.
make: stopped in /usr/home/seanc/go/src/github.com/hashicorp/nomad
11:01 seanc@hormesis % echo $GOPATH
/usr/home/seanc/go
```

And what's really bothering me is it works sometimes, like a few minutes later it will just work.  I've found that, for *whatever* reason, if I toggle between `bsdmake vet` and `gmake vet` that I can make forward progress, but I don't understand why this isn't 100% working or not.  It's like something is being pulled out of a `map` and the sort-order is non-deterministic, therefore the behavior is non-deterministic.

**Comment by kaey** [comment]
This looks like a totally different problem.
Please, file a new issue and try to find smallest possible repro. Thanks.

**Comment by gliptak** [comment]
I'm also seeing this (with the similar steps to reproduce):
```
$ mkdir /tmp/test
$ cd /tmp/test
$ export GOPATH=`pwd`
$ go get github.com/kardianos/govendor
$ ./bin/govendor init
Error: Package "/tmp/test" not a go package or not in GOPATH.
$ go env
GOARCH="amd64"
GOBIN=""
GOEXE=""
GOHOSTARCH="amd64"
GOHOSTOS="linux"
GOOS="linux"
GOPATH="/tmp/test"
GORACE=""
GOROOT="/usr/lib/go-1.6"
GOTOOLDIR="/usr/lib/go-1.6/pkg/tool/linux_amd64"
GO15VENDOREXPERIMENT="1"
CC="gcc"
GOGCCFLAGS="-fPIC -m64 -pthread -fmessage-length=0"
CXX="g++"
CGO_ENABLED="1"
```
Any pointers on what I'm doing wrong?
Thanks

**Comment by kardianos** [comment]
/tmp/test/src/test
Look up GOPATH def

**Comment by gliptak** [comment]
@kardianos Daniel, https://golang.org/doc/code.html#GOPATH guides to point ```GOPATH``` to the root of the workspace (as was done above). Is there a better practice to follow? Thanks

**Comment by kardianos** [comment]
You're misunderstanding how to use GOPATH I think.

Define a GOPATH

GOPATH looks like
```
/tmp/test
   bin/
   pkg/
   src/
       github.com/user/project
       mydomain.com/project
```

You don't put stuff directly in GOPATH, you put it in a directory under src, and the src dir is in GOPATH.

**Comment by gliptak** [comment]
Yes, I'm not quite seeing how this ```GOPATH``` variable works :(
Here is how ```/tmp/test``` looks after running the commands above:
```
/tmp/test/src/github.com/kardianos/govendor
/tmp/test/pkg/linux_amd64/github.com/kardianos/govendor
/tmp/test/bin
```
Could you offer the right commands to run ```govendor init``` for this tree?
Thanks

**Comment by kardianos** [comment]
```
cd /tmp/test/src
mkdir myproj
cd myproj
govendor init

```

**Comment by gliptak** [comment]
Thank you that worked.
So one would keep the main project also under ```src```?
Would ```govendor``` work for https://github.com/hashicorp/vault (seem to have different layout, but also has a vendor folder ...)?

**Comment by kardianos** [comment]
vault already uses govendor to manage deps. Now you only checkin the content `myproj` folder, not the whole tree. Your `.git` would be inside the `myproj` folder.

**Comment by gliptak** [comment]
OK, I cloned ```vault```, and running ```govendor status``` shows the same message ...
```
$ cd /tmp
$ git clone git@github.com:hashicorp/vault.git
$ export GOPATH=/tmp/vault/vault
$ cd /tmp/vault/vault
$ /tmp/test/bin/govendor status
Error: Package "/tmp/vault" not a go package or not in GOPATH.
govendor status
	Shows any packages that are missing, out-of-date, or modified locally (according to the
	checksum) and should be sync'ed.
```

**Comment by kardianos** [comment]
Go get github.com/hashicrp/vault

**Comment by gliptak** [comment]
Thank you. Placing packages into ```src/``` using the the long path does work. But I'm still unclear how one creates a new project utilizing ```vendor/``` ...

**Comment by RenEvo** [comment]
@kardianos it might be better to detect if the current directory **is** the GOPATH, that the message states must be in a package and not in the root GOPATH. If the current directory isn't in the GOPATH, the current message makes sense.

**Comment by gittycat** [comment]
Plague by this issue too. It's quite common to create simple standalone go programs in some nondescript location, eg: `/mywork/goapp.go`. Govendor shouldn't enforce the presence of a `src` dir.

**Comment by kaey** [comment]
@gittycat This is not a problem with govendor, this is a problem with go tool itself - vendor dir doesn't work outside of GOPATH.

**Comment by geraldstanje** [comment]
i get the same error. any solution yet?

**Comment by IgnacioCarbajoVallejo** [comment]
I don't if I have the same problem...

```
root@MyHostname:~/go/src/github.com/myUser/myApp # go env GOPATH
/root/go
```

```
root@MyHostname:~/go/src/github.com/myUser/myApp # govendor init
Error: Missing GOPATH. Check your environment variable GOPATH.
```

Any suggestions?

**Comment by RenEvo** [comment]
@geraldstanje after a few months working through a few apps I have realized that you have to update your brain a bit for working with GO as opposed to other languages.

`GOPATH` is essentially your workspace NOT your project where `<GOPATH>/src/<yourprojects>` is your project.

When creating your project, I recommend the following:

```
mkdir ./project
cd ./project
export GOPATH=$(pwd)
mkdir -p ./src/github.com/yourname/yourproject
cd ./src/github.com/yourname/yourproject
git init
govendor init
git commit -am "Initial"
```

From there you just get dependencies and work like normal from your projects directory.

**Comment by youwen21** [comment]
windows  right: gopath=D:\gopath    ,  wrong: gopath=D:\gopath\;  or D:\gopath;          
edit text
